src/libdrakvuf/drakvuf.c

```diff
@@ -169,6 +168,10 @@ void drakvuf_close(drakvuf_t drakvuf, const bool pause)
     drakvuf->json_kernel_path = NULL;
     g_free(drakvuf->json_wow_path);
     drakvuf->json_wow_path = NULL;
+
+    drakvuf_release_vmi(drakvuf);
```

Comment:
Well, unlocking if its not locked is also undefined behavior. So probably want to do a trylock before.

Reply:
I would say the lock should be taken as the first step in this function.

Reply:
Line 128

Reply:
Ah good, I did that already :)